File: Resoluções OJ/uri/1684.py
t = int(input())
for caso in range(t):
    n , m = map(int,input().split())
    f = {}
    for i in range(n):
        f[i] = 0
    for i in range(m):
        a , b = map(int,input().split())
        f[a] , f[b] = (f[a] + 1) , (f[b] + 1)
    ans = True
    for i in range(n):
        if (f[i] & 1):
            ans = False
            break
    print("Yes" if ans else "No")

File: Resoluções OJ/uri/1684.cpp
#include<bits/stdc++.h>
#define _ ios_base::sync_with_stdio(0);

using namespace std;

int f[10123],n,m,a,b,ans,t;

int main(){_
    for (cin>>t;t--;){
        cin>>n>>m;
        while (m--){
            cin>>a>>b;
            ++f[a] , ++f[b];
        }
        ans = 1;
        for (int i=0; i<n; i++){
            ans &= (!(f[i]&1));
            f[i] = 0;
        }
        cout<<(ans ? "Yes" : "No")<<endl;
    }
    return 0;
}
